herbert_core/classes/instrument_classes/instrument/detectors/@IX_detector_array/IX_detector_array.m

```diff
@@ -162,9 +162,36 @@
                             'Detpar structure must be a scalar structure')
                     elseif all( isfield(S, {'filename','filepath',...
                             'group','x2','phi','azim', 'width', 'height'}))
-                        obj.det_bank_ = IX_detector_bank ( ...
-                            S.group, S.x2, S.phi, S.azim, ...
-                            IX_det_TobyfitClassic (S.width, S.height));
+                        
+                        % get the default types and parameters for a detector - these should have been
+                        % set by the user before use here. 
+                        [dtype,dparms] = IX_det_abstractType.tobyfit_defaults();
+                        if strcmp(dtype,'IX_det_He3tube')
+                            obj.det_bank_ = IX_detector_bank ( ...
+                                S.group, S.x2, S.phi, S.azim, ...
+                                IX_det_He3tube (S.width, S.height, ...
+                                dparms.wall, dparms.p));
+                        elseif strcmp(dtype,'IX_det_TobyfitClassic')
+                            obj.det_bank_ = IX_detector_bank ( ...
+                                S.group, S.x2, S.phi, S.azim, ...
+                                IX_det_TobyfitClassic (S.width, S.height));
+                        elseif isempty(dtype)
+                            %{
+                            % omitting this warning for the moment as it
+                            % causes problems with
+                            % test_admin/upgrade_file_formats, but it may be
+                            % useful to reinstate it at some stage
+                            warning("HORACE:IX_detector_array:empty detector type", ...
+                                    "probably on load from detpar");
+                            %}
+                            obj.det_bank_ = IX_detector_bank ( ...
+                                S.group, S.x2, S.phi, S.azim, ...
+                                IX_det_TobyfitClassic (S.width, S.height));
+                        else
+                            error('HORACE:IX_detector_array:invalid_type', ...
+                                  'unsupported detector type');
+                        end
```

Comment:
Could use a `switch` here rather than an `if` chain.
```suggestion
                        switch(dtype)
                          case 'IX_det_He3tube'
                            obj.det_bank_ = IX_detector_bank ( ...
                                S.group, S.x2, S.phi, S.azim, ...
                                IX_det_He3tube (S.width, S.height, ...
                                dparms.wall, dparms.p));
                          case 'IX_det_TobyfitClassic'
                            obj.det_bank_ = IX_detector_bank ( ...
                                S.group, S.x2, S.phi, S.azim, ...
                                IX_det_TobyfitClassic (S.width, S.height));
                          case ''
                            %{
                            % omitting this warning for the moment as it
                            % causes problems with
                            % test_admin/upgrade_file_formats, but it may be
                            % useful to reinstate it at some stage
                            warning("HORACE:IX_detector_array:empty detector type", ...
                                    "probably on load from detpar");
                            %}
                            obj.det_bank_ = IX_detector_bank ( ...
                                S.group, S.x2, S.phi, S.azim, ...
                                IX_det_TobyfitClassic (S.width, S.height));
                          otherwise
                            error('HORACE:IX_detector_array:invalid_type', ...
                                  'unsupported detector type');
                        end
```

Reply:
yes, one could